system/jhtree/keybuild.cpp

```diff
@@ -74,6 +74,33 @@ class CRC32EndHT : public CRC32HT
     virtual bool matchesFindParam(const void *et, const void *fp, unsigned) const { return *(offset_t *)((const CRC32HTE *)et)->queryEndParam() == *(offset_t *)fp; }
 };
 
+interface IIndexCompressor : public IInterface
+{
+    virtual CompressionType queryCompressionType() const = 0;
+    virtual const char *queryName() const = 0;
+    virtual CWriteNode *createNode(offset_t _fpos, CKeyHdr *_keyHdr, bool isLeafNode) const = 0;
+};
+
+class PocIndexCompressor : public CInterfaceOf<IIndexCompressor>
+{
+    virtual CompressionType queryCompressionType() const override { return SplitPayloadLeaf; };
```

Comment:
I think you have used the same enumeration for the subtype as the compression type.  That isn't quite what I expected.  Is this function used?  If not it should be deleted, or changed to a const char * return?  I don't know what I would use for my compression code.

How should the subtype relate to the node type?  In my case I have leaf and branch nodes both of which can use inplace-keys.  There might be 2 or 3 implementations of the leaf nodes.  I think what I had imagined was a separate enumeration for each node type.  So

enum BranchSubType : byte { BranchLegacy, BranchInplace }
enum LeafSubType : byte ( LeafLegacy, LeafSlitPayload, LeafInplaceLzw, LeafInplaceDeflate, LeafInplaceX }

The enumerations could also have entries for the old variants e.g. LeafVariable LeafRowCompressed if it made the code any easier.